### DISCUSSION TITLE
Some questions about GapReports

### DISCUSSION THREAD

**Discussion by sysyboom** [SysBioChalmers/RAVEN#449] [Help me!]
Hi,
 I face some challenges when using your software RAVEN2.0 for fungal model reconstruction.Firstly, I built a combined model from KEGG and Metacyc by using the function, and after adding nutrient sources to the model, I did a gapReport, but the error "Matrix index is out of range for deletion" came from like this" Error in removeReactions (line 74)  reducedModel.equations(indexesToDelete,:)=[];" In the same time, I ran GapReports with the model built by Kegg only, it works.
    The problems are these： 1) Whether gapReport function input only require the kegg model or not? If not,What might be the source of this error？
                                          2)Does the RAVEN2.0 provide a Function to convert the metacyc model to the keggmodel?
                                       
     Looking forward to your reply！PS：Attachment is the error details.
![Error details](https://user-images.githubusercontent.com/114929833/200829059-1c07ff9c-2900-4d50-8dcb-1b091c1d585c.jpg)

**Comment by edkerk** [comment]
1. gapReport _should_ work on any model, such an error message _should_ not occur. What happens if you run gapReport on the MetaCyc-only model? Then we can now if the problem is in the MetaCyc model, or in the function that combines the two models.
2. RAVEN does not have a function to fully convert the models, not all metabolites and reactions that are on MetaCyc are also on KEGG (possibly also vice versa, but I'm not certain about that), so such a function would not be able to convert the whole model.

**Reply by sysyboom** [reply]
Thanks for your reply! I meet the same error message when I run gapReport on the MetaCyc-only model ,Maybe I should check my model~

**Comment by dimwas** [comment]
Good day to you,

I have a problem with gapReports as well. I tried to run gapReports for my model, and got the following error (immediately after beginning of Metabolite connectivity section):
```
Unable to perform assignment because the left and right sides have a different number of elements.

Error in getIndexes (line 79)
            indexes(i)=index;

Error in haveFlux (line 40)
indexes=getIndexes(model,rxns,'rxns');

Error in checkProduction (line 65)
canProduce=haveFlux(model,10^-5,addedRxns);

Error in gapReport (line 83)
[notProducedMets, ~, neededForProductionMat,minToConnect]=checkProduction(model,true,model.comps,false);
```
I tried to run gapReport for template models, with the same result. One of the template models is the one for [P. chrysogenum
](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002980). The only one model that did not throw an error on this place was the yeast9 model, but there was no output after starting of Metabolite connectivity section either for more than 1 hour, so I decided to interrupt the program. 

What could be the reason for this error?

**Reply by dimwas** [reply]
OK, I have found the problem. In my model, I had duplicated metabolites, therefore `getIndexes` threw out an error.

**Reply by edkerk** [reply]
Thank you for reporting the solution to your problem, I will refactor the code to throw a more informative error message.
